### PR TITLE
Add Typescript test & example

### PR DESCRIPTION
**Pull request by azizhk**
Followup on #613

### CODE REVIEW

File: examples/typescript/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "typescript-example",
+  "version": "1.0.0",
+  "license": "MIT",
+  "scripts": {
+    "test": "tsc --noEmit"
```
Comment (azizhk) [inline examples/typescript/package.json]: Need to add this command to travis tasks. Will need help here.

Comment (radex) [reply examples/typescript/package.json]: this is the important bit: https://github.com/Nozbe/WatermelonDB/blob/master/.travis.yml#L31 -- so if you add to the root package.json something like `test:typescript = ... cd examples/... yarn testst` and then to `ci:check` add `previous commands && yarn test:typescript` that śhould work

Comment (azizhk) [reply examples/typescript/package.json]: Done.

**Comment by radex** [comment]
@azizhk is this ready for review?

**Comment by azizhk** [comment]
Yes this is ready for review. I'll just squash all commits.

**Comment by radex** [comment]
@azizhk don't squash. please add a note to CHANGELOG

**Comment by radex** [comment]
thanks!

**Pull request merged**